app/services/clients/stripe/stripe_client.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+// NPM dependencies
+const Stripe = require('stripe')
+const ProxyAgent = require('https-proxy-agent')
+
+// Local dependencies
+const StripeBankAccount = require('./stripeBankAccount.model')
+
+// Constants
+const STRIPE_LIVE_API_KEY = process.env.STRIPE_LIVE_API_KEY
+const STRIPE_TEST_API_KEY = process.env.STRIPE_TEST_API_KEY
+const STRIPE_HOST = process.env.STRIPE_HOST
+const STRIPE_PORT = process.env.STRIPE_PORT
+
+const getStripe = function getStripe (gatewayAccount) {
+  const apiKey = (gatewayAccount.type === 'test') ? STRIPE_TEST_API_KEY : STRIPE_LIVE_API_KEY
+  const stripe = Stripe(apiKey)
+
+  if (process.env.http_proxy) {
+    stripe.setHttpAgent(new ProxyAgent(process.env.http_proxy))
+  }
+
+  // only expect host and port environment variables to be set when running tests
+  if (STRIPE_HOST) {
+    stripe.setHost(STRIPE_HOST)
+  }
+  if (STRIPE_PORT) {
+    stripe.setPort(STRIPE_PORT)
+  }
+
+  return stripe
+}
+
+module.exports = {
+  updateBankAccount: function (gatewayAccount, stripeAccountId, body) {
+    const bankAccount = new StripeBankAccount(body)
+    const stripe = getStripe(gatewayAccount)
```

Comment:
At the moment this initialises a new Stripe object per request to update a bank account, self service doesn't have the same scale requirements as other using facing services however this means that for 10 requests we make 10 (relatively large) objects, for 100 we make 100. 

My understanding is that the Stripe SDK returns a fairly large object that is able to make requests based on its configuration -- that shouldn't change after the server has started. 

If we have to connect to both Live and Test accounts we would need two calls to `Stripe`, otherwise we may only need one. 

For example: 
```js
const stripe = require('stripe')(STRIPE_API_KEY)

stripe.setHttpAgent() //...
stripe.setHost() //...
stripe.setPort() //...

const updateBankAccount = function updateBankAccount () { 
  return stripe.accounts.update() //...
}

module.exports = { updateBankAccount }
```

Reply:
Makes sense, thanks 👍